aea/helpers/dialogue/base.py

```diff
@@ -567,30 +615,80 @@ def _basic_rules(self, message: Message) -> bool:
         target = message.target
         performative = message.performative
 
-        if self.last_message is None:
-            result = (
-                dialogue_reference[0] == self.dialogue_label.dialogue_reference[0]
-                and message_id == Dialogue.STARTING_MESSAGE_ID
-                and target == Dialogue.STARTING_TARGET
-                and performative in self.rules.initial_performatives
-            )
-        else:
+        if self.last_message is None:  # Initial message
```

Comment:
There are a lot of lines here in one method; suggest breaking it up and having some more helper methods (maybe content of each block

Reply:
Done